### PR TITLE
Bump go version to 1.21

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

Bumps the go version to v1.21 and updates the gh actions workflows accordingly as well. 

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -39,5 +39,8 @@ issues:
     - path: "(.+)_test.go"
       linters:
         - gosec
+    - path: "internal/config/config.go"
```
Comment (sergiught) [inline .golangci.yml:42]: Go sec is reporting hardcoded credentials on:

```
const providerName = "Terraform-Provider-Auth0"
```

**Review by Widcket — APPROVED** [review]

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Set up Go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v4
```
Comment (Widcket) [inline .github/workflows/e2e-tests.yml:18]: Should we pin this one using a commit SHA as well?

Comment (sergiught) [reply .github/workflows/e2e-tests.yml:18]: According to semgrep not, as it's not considered a 3rd party action.

File: .golangci.yml (hunk 1 above)
Comment (willvedd) [reply .golangci.yml:42]: Can we add inline w/ code to prevent exclusion of the entire file?

**Review by willvedd — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (sergiught) [reply .golangci.yml]: Updated in [`f81d81a` (#789)](https://github.com/auth0/terraform-provider-auth0/pull/789/commits/f81d81ab0e2f1a45ea904b2bcae81040b0a65d57)

**Pull request merged**